sdk/core/core-tracing/src/tracingClient.ts

```diff
@@ -64,14 +65,14 @@ export function createTracingClient(options: TracingClientOptions): TracingClien
     operationOptions: Options,
     callback: Callback,
     spanOptions?: TracingSpanOptions
-  ): Promise<ReturnType<Callback>> {
+  ): Promise<AwaitedLike<ReturnType<Callback>>> {
     const { span, updatedOptions } = startSpan(name, operationOptions, spanOptions);
     try {
       const result = await withContext(updatedOptions.tracingOptions!.tracingContext!, () =>
         Promise.resolve(callback(updatedOptions, span))
       );
       span.setStatus({ status: "success" });
-      return result;
+      return result as ReturnType<typeof withSpan>;
```

Comment:
This is not needed in TS 4.4 [see playground link](https://www.typescriptlang.org/play?ts=4.4.4#code/IYZwngdgxgBAZgV2gFwJYHsIwO6uQCwGUAHYCAHgCgYaYBhYAG0YCNgoBrGAUwA9luEACYgYACmq0AlDAC8APhgAlbsgQAnCABUwxbuQbM2neZPkTJNKE1bsOALno3jHADSSpjgArr0AW1QQfRU1TR09A2c7eUUAb0sYdVUNLGsjOzEpAG5JAF9KSWRdbhgAQWxgPG4hci1FWRgtHn5BERgIBGYYAB8YJCFuOFQIaskAfkaYAHopmBA9KFQmGGsg+HR1GAADDq7e-sHh6q2cfEF29GQYYe2AWluQZHVUKGQAOU7GOjPOEBO-dADSSOJp8ATCUToFgAK24r2msy2wAqeBOmEYYD6EGw6mAxEhMLhVyKelEuAIMGAKxswBYjBKBEEADoYG9MLcobD4STuKJgEkLlckDi8XohEzxpMwa1RLEYIyIGJMIhGENmNVHMM4NxNgAxTyUiCY3II+VnCC0+muZpPdjIUSM+CodSPSnqADmCD8gmJ6G2CsyWwSE11Npl4gAbkwENxNRBtZsAGoyBSGzEza5wM0lfme70QX3+80nQLU5iW7jWsG414Os5uvM+hI0CblSoCGqJxQZpJQDQgVAR7gYrEi4jZmBRxgx5swRwjIebDOO3Nen3yv1bBUnCqiCCXMuMCsS2hzyYZ1f5q67wWGmCconAxpZU37iC3BUVgo0UCQWCIFAMCwckiFICA2yqGoEkMWxODDCFxG-aQ5EUEIUnCfQYJcUwaHMBI0lghwnHSTh3BoA0fH8QJ9AgjtyDQsJikiEiOBiOJ8MwV0khATorgaZF2xgSiAiCJluPQRghzEAiXEybIEgzIQ-RAfwSmIdBHluYhfCgXkBwgd0cDwfAJ243iEiSUIsDMxhkByGh8kkECSDIMRMhQmAAFZ5JoZywLclNFGE6ixN5CSpO8qQnOMlzwJRDsAo87zooIWLaOqRLU2C0TxMk7gxEioA) but for reasons unknown to me TS4.2 infers the returntype of this as `unknown`

Reply:
Thanks for sharing the playground!